pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -1435,6 +1435,61 @@ class Foo{
 ]]>
         </example>
     </rule>
+    <rule name="UnnecessaryVarargsArrayCreation"
+          language="java"
+          since="7.1.0"
+          message="Unnecessary explicit varargs array creation"
+          class="net.sourceforge.pmd.lang.java.rule.bestpractices.UnnecessaryVarargsArrayCreationRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#unnecessaryvarargsarraycreation">
+        <description>
+            Reports explicit array creation when a varargs is expected.
+            For instance:
+            ```java
+            Arrays.asList(new String[] { "foo", "bar", });
+            ```
+            can be replaced by:
+            ```java
+            Arrays.asList("foo", "bar");
+            ```
+
+            This rule also reports such array creations when they are confusing, because the array is a subtype of the component type of the expected array type.
```

Comment:
Note that ideally this would be in another rule in errorprone, because it's a more severe type of warning. But the implementation is really easy to grab...